### PR TITLE
Fix dead link in MySQL Provider page

### PR DESCRIPTION
**Pull request by fritz-astronomer**
Doc only change, fixing a dead link

### CODE REVIEW

File: docs/apache-airflow-providers-mysql/index.rst [hunk 1]
```diff
@@ -39,13 +39,13 @@ Content
     :hidden:
     :caption: System tests
 
-    System Tests <_api/tests/system/providers/mysql/index>
+    System Tests <_api/tests/providers/mysql/index>
```
Comment (fritz-astronomer) [inline docs/apache-airflow-providers-mysql/index.rst]: not sure if this one is correct 🤔 would appreciate confirmation

Comment (uranusjr) [reply docs/apache-airflow-providers-mysql/index.rst]: This one mentions system tests specifically so the `system` one is (probably) correct.

File: docs/apache-airflow-providers-mysql/index.rst [hunk 2]
```diff
@@ -39,13 +39,13 @@ Content
...
 
 .. toctree::
     :maxdepth: 1
     :caption: Resources
 
-    Example DAGs <https://github.com/apache/airflow/tree/tests/system/providers/mysql>
+    Example DAGs <https://github.com/apache/airflow/blob/main/tests/providers/mysql/>
```
Comment (uranusjr) [inline docs/apache-airflow-providers-mysql/index.rst]: Same here, should point to `tests/system/providers/mysql` instead.

File: docs/apache-airflow-providers-mysql/index.rst (hunk 1 above)
Comment (potiuk) [reply docs/apache-airflow-providers-mysql/index.rst]: Yes @fritz-astronomer  - I think what we need to fix is the "old" link generated directly in the generated [arifow-site](https://github.com/apache/airflow-site/tree/main/docs-archive/apache-airflow-providers-mysql) instead. 

This one should stay as it is.

**Comment by fritz-astronomer** [comment]
got busy, closing this til I can think about it again, unless anyone else can pick it up

**Review by uranusjr — APPROVED** [review]

**Pull request merged**